### PR TITLE
grant usage on tables in pld schemas

### PR DESCRIPTION
**Pull request by ondrajodas**
Jira - https://keboola.atlassian.net/browse/CM-665

### CODE REVIEW

File: src/Migrate.php [hunk 1]
```diff
@@ -1379,11 +1379,13 @@ private function getDataToRemove(string $role): array
...
     }
 
-    private function canCloneTable(string $database, string $schema, string $table): bool
+    private function canCloneTable(string $mainRole, string $database, string $schema, string $table): bool
     {
 
         $sqlTemplate = 'SELECT max("_timestamp") as "maxTimestamp" FROM %s.%s.%s';
 
+        $currentRole = $this->destinationConnection->getCurrentRole();
+        $this->destinationConnection->useRole($mainRole);
```
Comment (pivnicek) [inline src/Migrate.php]: ah, tohle mi trval chvili to pochopit ale uz to chapu :-)

**Review by pivnicek — DISMISSED** [review]
nic sileniho nevidim

File: src/Migrate.php [hunk 2]
```diff
@@ -1429,20 +1432,20 @@ private function grantUsageToOldTable(
...
+                'grant usage on schema %s.%s to role %s;',
                 Helper::quoteIdentifier($database . '_OLD'),
-                $role
+                Helper::quoteIdentifier($schema),
+                Helper::quoteIdentifier($role)
             ),
             sprintf(
-                'grant usage on table %s in schema %s.%s to role %s;',
-                Helper::quoteIdentifier($table),
+                'grant select on table %s.%s.%s to role %s;',
```
Comment (pivnicek) [inline src/Migrate.php:1444]: j, I thought that usage maybe wouldn't be enough

**Review by pivnicek — APPROVED** [review]
lgtm

File: src/Migrate.php (hunk 2 above)
Comment (ondrajodas) [reply src/Migrate.php:1444]: why not?

**Pull request merged**